### PR TITLE
US1282 Decrease metadata writes

### PR DESCRIPTION
**Pull request by jkryl**
Replace redundant_metadata all value by none, which should be the new
default. We cannot extend the set of existing values (all and most),
because of compatibility reason.

Fix minor ref count leak on mgmt connection due to iscsi target quering
zvol which is not associated with mgmt connection.

Brief performance testing shows that iops increased by ~20% when measured
random read-write by fio compared to redundant_metadata=all. I haven't seen
much difference compared to redundant_metadata=most. That might have been
caused by relatively small size of zvol that I used for testing (7G).

### CODE REVIEW

File: cmd/zrepl/mgmt_conn.c [hunk 1]
```diff
@@ -821,13 +821,25 @@ process_message(uzfs_mgmt_conn_t *conn)
 		strncpy(zvol_name, payload, payload_size);
 		zvol_name[payload_size] = '\0';
 
-		if (((zinfo = uzfs_zinfo_lookup(zvol_name)) == NULL) ||
-		    (zinfo->mgmt_conn != conn)) {
```
Comment (vishnuitta) [inline cmd/zrepl/mgmt_conn.c:804]: this condition makes sure that commands are received on correct connection, and no issue happened in replicas communicating with target. Removing this condition can lead to any issues due to any issues with replicas or targets.
Its better to have this condition.

Comment (vishnuitta) [reply cmd/zrepl/mgmt_conn.c:804]: oh sorry.. got to see this in next line.. :) :+1:

Comment (vishnuitta) [reply cmd/zrepl/mgmt_conn.c:804]: and also the refcount issue.. thanks for fixing this @jkryl

**Review by vishnuitta — COMMENTED** [review]
Are we reducing by one copy of metadata? And, can we have some testcases automated for this?

**Comment by jkryl** [comment]
All I can test is that redundant_metadata is set to none by default. I don't see an easy way how I could test that particular block was written one time instead of two times.. I will add the test for none.

**Comment by satbirchhikara** [comment]
Hi Jan,

Can we see if spa_max_replication_override this variable make sense to reduce metadata copy to one ?

Thanks & Regards,
Satbir Singh

**Comment by jkryl** [comment]
Hi Satbir, good catch! This tunable is for MOS (meta-objset). If we want to optimize writing to this dataset, we need to change that tunable too. MOS is written each time when writing new blocks in zvol to update the root block pointer so it is important optimization. Currently I set it to 1.

@vishnuitta : what are your thoughts about this? Isn't it too dangerous? The problem is that if MOS gets corrupted, then we lose access to all zvols in a pool. Do we still want to maintain some degree of redundancy at this level (i.e. setting the tunable to 2)?

**Comment by jkryl** [comment]
As per our discussion during standup meeting, I reverted the change of MOS data block redundancy.

**Review by vishnuitta — APPROVED** [review]
changes are good.
